sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -17,37 +17,121 @@
+            (isEqual(firstFourBytesArray, new int[] {0x49, 0x49, 0x2a, 0x0}))
+                // big-endian
+                || (isEqual(firstFourBytesArray, new int[] {0x4d, 0x4d, 0x0, 0x2a}))) {
+            return ContentType.IMAGE_TIFF;
+        } else {
+            throw LOGGER.logExceptionAsError(new IllegalArgumentException(
+                "Content type could not be detected. Should use other overload API that takes content type."));
+        }
     }
 
-    private static ByteBuffer toByteArray(InputStream in) {
-        try {
-            ByteArrayOutputStream os = new ByteArrayOutputStream();
-            byte[] buffer = new byte[1024];
-            int len;
+    /**
+     * Creates a Flux of ByteBuffer, with each ByteBuffer wrapping bytes read from the given
+     * InputStream.
+     *
+     * @param inputStream InputStream to back the Flux
+     * @return Flux of ByteBuffer backed by the InputStream
+     */
+    public static Flux<ByteBuffer> toFluxByteBuffer(InputStream inputStream) {
+        Pair pair = new Pair();
+        return Flux.just(true)
+            .repeat()
+            .map(ignore -> {
+                byte[] buffer = new byte[BYTE_BUFFER_CHUNK_SIZE];
+                try {
+                    int numBytes = inputStream.read(buffer);
+                    if (numBytes > 0) {
+                        return pair.buffer(ByteBuffer.wrap(buffer, 0, numBytes)).readBytes(numBytes);
+                    } else {
+                        return pair.buffer(null).readBytes(numBytes);
+                    }
+                } catch (IOException ioe) {
+                    throw LOGGER.logExceptionAsError(new RuntimeException(ioe));
+                }
+            })
+            .takeUntil(p -> p.readBytes() == -1)
+            .filter(p -> p.readBytes() > 0)
+            .map(Pair::buffer);
+    }
+
+    private static class Pair {
+        private ByteBuffer byteBuffer;
+        private int readBytes;
 
-            // read bytes from the input stream and store them in buffer
-            while ((len = in.read(buffer)) != -1) {
-                // write bytes from the buffer into output stream
-                os.write(buffer, 0, len);
+        ByteBuffer buffer() {
+            return this.byteBuffer;
+        }
+
+        int readBytes() {
+            return this.readBytes;
+        }
+
+        Pair buffer(ByteBuffer byteBuffer) {
+            this.byteBuffer = byteBuffer;
+            return this;
+        }
+
+        Pair readBytes(int cnt) {
+            this.readBytes = cnt;
+            return this;
+        }
+    }
+
+    /**
+     * Compare if a byte value equals to a hex type value.
+     *
+     * @param byteValueList An array of byte type values.
+     * @param hexValueList An array of hex type values.
+     * @return true if two type's values are equal in unsigned int comparision, otherwise return false.
+     */
+    private static boolean isEqual(byte[] byteValueList, int[] hexValueList) {
```

Comment:
I don't feel this abstraction is really worth the overhead, both in maintenance and performance.